vcap_utils/vcap_utils/backends/base_openvino.py

```diff
@@ -35,34 +35,112 @@ def __init__(self, model_xml: bytes,
 
         self.ie = ie_core or IECore()
 
-        # Find the optimal number of InferRequests for this device
-        supported_metrics = self.ie.get_metric(
-            device_name, _SUPPORTED_METRICS)
-        if _RANGE_FOR_ASYNC_INFER_REQUESTS in supported_metrics:
-            low, high, _ = self.ie.get_metric(
-                device_name, _RANGE_FOR_ASYNC_INFER_REQUESTS)
-            # Cap the n_requests, because sometimes high_n crashes the system
-            # TODO(Alex): Figure out _why_ hddl crashes when set to 'high'
-            n_requests = max(0, min(low * 2, high))
-        else:
-            # Use the devices default
-            n_requests = 0
+        # If the device is a MULTI device, use the default num_requests
+        is_multi_device = device_name.lower().startswith("multi")
+
+        num_requests = 0
+        if not is_multi_device:
+            # Find the optimal number of InferRequests for this device
+            supported_metrics = self.ie.get_metric(
+                device_name, _SUPPORTED_METRICS)
+            if _RANGE_FOR_ASYNC_INFER_REQUESTS in supported_metrics:
+                low, high, _ = self.ie.get_metric(
+                    device_name, _RANGE_FOR_ASYNC_INFER_REQUESTS)
+                # Cap the num_requests, because setting it too high can crash
+                # TODO(Alex): Figure out _why_ hddl crashes when set to 'high'
+                num_requests = max(0, min(low * 2, high))
 
         self.net: IENetwork = self.ie.read_network(
             model=model_xml,
             weights=weights_bin,
             init_from_buffer=True)
 
-        self.exec_net: ExecutableNetwork = self.ie.load_network(
-            network=self.net,
-            device_name=device_name,
-            num_requests=n_requests)
+        try:
+            self.exec_net: ExecutableNetwork = self.ie.load_network(
+                network=self.net,
+                device_name=device_name,
+                num_requests=num_requests)
+        except RuntimeError as e:
+            if device_name == "CPU":
+                # It's unknown why this would happen when loading
+                # onto CPU, so we re-raise the error.
+                raise e
+
+            # This error happens when trying to load onto HDDL or MYRIAD, but
+            # somehow the device fails to work. In these cases, we try again
+            # but load onto 'CPU', which is a safe choice.
+            msg = f"Failed to load {self.__class__} onto device " \
+                  f"{device_name}. Error: '{e}'. " \
+                  f"Trying again with device_name='CPU'"
+            logging.critical(msg)
```

Comment:
Nit: I think `error` might be the more fitting log level here, since this is an error we can recover from and have the system run normally. I think of critical errors as being ones that make the system crash or be unstable.

Reply:
> Nit: I think `error` might be the more fitting log level here, since this is an error we can recover from and have the system run normally. I think of critical errors as being ones that make the system crash or be unstable.

Good idea!